### DISCUSSION TITLE
Proposal: maintain release branches

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#2208] [Ideas] [answered]
The method we use to work on our modules is to pull in oqtane.framework as a submodule into our oqtane module solution.  This has the benefit of a single solution build.  The downside is that .gitmodules file can only specify the branch and not the commit.  So as new versions of oqtane framework is released, we need to be careful when cloning the repo to not jump-ahead accidently. 

Maintaining release branches called release/[tag] would allow us to pin our .getmodules to a set version.

I suppose the alternative to this approach would be to fork oqtane.framework and take on this administrative chore ourselves.  But I'm lazy and it would be fun to share this single-solution via submodule method with the community.  Now we use the sub-module method with the caveat that you "need to be careful".

**Comment by sbwalker** [comment]
@markdav-is I believe what you are looking for already exists. Whenever a release is done in Github it applies a "tag" to all of the assets which are part of the release. This allows you to reference the complete set of code at that point in time. If you go to the Releases area https://github.com/oqtane/oqtane.framework/releases and select the Tags option you will see a list of all releases - and every release has a unique Url ie. https://github.com/oqtane/oqtane.framework/releases/tag/v3.1.2 

![image](https://user-images.githubusercontent.com/4840590/169827184-2bc611fd-4d2d-49d1-bae7-8291e922a5be.png)

**Reply by markdav-is** [reply]
That's true!  We could pull-by-tag and have a sibling repo.  That's how we started with Oqtane.  The limitation is really how .gitmodules is implemented and how you can't pull a sub-module by commit (easily).  We moved to sub-modules to help ease the path to Oqtane module development in a single solution.  The work-around would be release branches, but I know it's a big lift for this edge-case.  I think what I'll do is fork oqtane and maintain release branches in my own github orgs.  Then I can get those release branches I so excited about and you don't need to do extra work. 

I could also submit a PR to upgrade gitmodules :)  https://git-scm.com/docs/gitmodules

**Comment by markdav-is** [comment, marked as answer]
If anyone else needs .submodules to point to a particular branch, I'm maintaining release branches here: https://github.com/DarkMatterOregon/oqtane.framework/branches

**Reply by hishamco** [reply]
The link is broken

**Reply by leigh-pointer** [reply]
@hishamco works for me,

**Reply by hishamco** [reply]
It's ok now ;)
